x-pack/legacy/plugins/snapshot_restore/public/app/sections/home/recovery_list/recovery_table/shards_table.tsx

```diff
@@ -0,0 +1,252 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import {
+  EuiBasicTable,
+  EuiProgress,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiLoadingSpinner,
+  EuiText,
+  EuiSpacer,
+  EuiToolTip,
+} from '@elastic/eui';
+import { SnapshotRecovery, SnapshotRecoveryShard } from '../../../../../../common/types';
+import { useAppDependencies } from '../../../../index';
+import { formatDate } from '../../../../services/text';
+
+interface Props {
+  shards: SnapshotRecovery['shards'];
+}
+
+export const ShardsTable: React.FunctionComponent<Props> = ({ shards }) => {
+  const {
+    core: { i18n },
+  } = useAppDependencies();
+  const { FormattedMessage } = i18n;
+
+  const columns = [
+    {
+      field: 'id',
+      name: i18n.translate('xpack.snapshotRestore.recoveryList.shardTable.indexColumnTitle', {
+        defaultMessage: 'ID',
+      }),
+      width: '40px',
+      render: (id: SnapshotRecoveryShard['id'], shard: SnapshotRecoveryShard) =>
+        shard.primary ? (
+          <EuiFlexGroup alignItems="center" gutterSize="xs">
+            <EuiFlexItem grow={false}>{id}</EuiFlexItem>
+            <EuiFlexItem grow={false}>
+              <EuiToolTip
+                position="right"
+                content={
+                  <FormattedMessage
+                    id="xpack.snapshotRestore.recoveryList.shardTable.primaryTooltipLabel"
+                    defaultMessage="Primary"
+                  />
+                }
+              >
+                <strong>
+                  <FormattedMessage
+                    id="xpack.snapshotRestore.recoveryList.shardTable.primaryAbbreviationText"
+                    defaultMessage="P"
```

Comment:
@chiweichang is this abbreviation translatable?

Reply:
To not block the PR from getting merged, we can merge this as is for now and get back to it if it needs an update.

Reply:
Sorry for the delayed reply; just catching up after being OOO. @Bamieh, were you referring to `defaultMessage="P"`? Then yeah, since we aren't able to see these in context, there's no way for the translators to know what this stands for. I would recommend spelling it out.

Reply:
@chiweichang The table information here is very dense, hence I am using an abbreviation with a tooltip - `P` meaning `Primary`:

![image](https://user-images.githubusercontent.com/1965714/60470030-b8305e00-9c13-11e9-95a4-5ae7c720fb16.png)

I've tried to allude to the meaning of this string with the naming of its ID, `primaryAbbreviationText`. Please let me know if is another way to provide context to translators?

Reply:
Hi @jen-huang, thanks for the reply. You should be able to save notes as part of the metadata; I would recommend noting there to provide context. I've seen other teams do that for their strings, and it's helped the translators immensely.

Reply:
@chiweichang I see, I will look into that. Thanks!